src/main/java/uk/gov/hmcts/ccd/domain/service/supplementarydata/rolevalidator/DefaultUserRoleValidator.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.ccd.domain.service.supplementarydata.rolevalidator;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.service.common.CaseAccessService;
+
+@Service
+@Qualifier("default")
+public class DefaultUserRoleValidator implements UserRoleValidator {
```

Comment:
this class could be renamed to be more generic so we can reuse it for other endpoints with similar access requirements. We could call it for example `EndpointAccessService` with a method `isAccessAllowed`. Or something similar

Reply:
Renamed